File: wlr/OGUR/OGUR/Items/Inventory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Creatures;
using OGUR.Entities;
using OGUR.Items;
using OGUR.Management;
using OGUR.Text;

namespace OGUR.Items
{
    public class Inventory
    {
        Dictionary<GenericItem,int> m_contents = new Dictionary<GenericItem, int>(); 
        private ICreature m_parent;

        public Inventory(ICreature parent)
        {
            m_parent = parent;
        }
        
        public ICreature GetParent()
        {
            return m_parent;
        }

        public GenericItem GetNonZeroEntry()
        {
            return m_contents.Keys.Where(k=>m_contents[k]>0).FirstOrDefault();
        }

        public void Add(GenericItem item)
        {
            if(Contains(item))
            {
                m_contents[item]++;
            }
            else
            {
                m_contents.Add(item, 1);
            }
        }

        private readonly Dictionary<GenericItem, int> m_itemResult = new Dictionary<GenericItem, int>();
        public Dictionary<GenericItem, int> GetItems(int iClass)
        {
            m_itemResult.Clear();
            foreach (var key in m_contents.Keys)
            {
                if (key.GetItemClass() == iClass)
                {
                    m_itemResult.Add(key, m_contents[key]);
                }
            }
            //return m_contents.Where(element => element.Key.GetItemClass() == iClass).ToDictionary(element => element.Key, element => element.Value);
            return m_itemResult;
        }

        public void Remove(GenericItem item)
        {
            if(Contains(item))
            {
                m_contents[item]--;
                if(m_contents[item]<=-1)
                {
                    m_contents.Remove(item);
                }
            }
        }

        public bool Contains(GenericItem item)
        {
            return m_contents.ContainsKey(item);
        }

        public int GetItemCount(GenericItem item)
        {
            if(m_contents.Keys.Contains(item))
            {
                return m_contents[item];
            }
            return 0;
        }

        public void DropAll()
        {
            foreach(var item in m_contents.Keys)
            {
                while(m_contents[item]>0)
                {
                    EntityManager.AddObject(new GenericItem(item,m_parent.GetLocation()));
                    Remove(item);
                }
            }
        }

        float _nonZeroResult;
        public float NonZeroCount()
        {
            _nonZeroResult = 0;
            foreach(var item in m_contents.Where(c=>c.Value>0))
            {
                _nonZeroResult += item.Value;
            }
            return _nonZeroResult;
        }
    }
}

File: proj/src/aigilas/items/Inventory.java
package aigilas.items;

import java.util.HashMap;

import spx.entities.EntityManager;
import aigilas.creatures.ICreature;

public class Inventory {
	HashMap<GenericItem, Integer> _contents = new HashMap<GenericItem, Integer>();
	private ICreature _parent;

	public Inventory(ICreature parent) {
		_parent = parent;
	}

	public ICreature GetParent() {
		return _parent;
	}

	public GenericItem GetNonZeroEntry() {
		for (GenericItem key : _contents.keySet()) {
			if (_contents.get(key) > 0) {
				return key;
			}
		}
		return null;
	}

	public void Add(GenericItem item) {
		if (Contains(item)) {
			_contents.put(item, _contents.get(item) + 1);
		}
		else {
			_contents.put(item, 1);
		}
	}

	private HashMap<GenericItem, Integer> _itemResult = new HashMap<GenericItem, Integer>();

	public HashMap<GenericItem, Integer> GetItems(ItemClass iClass) {
		_itemResult.clear();
		for (GenericItem key : _contents.keySet()) {
			if (key.GetItemClass() == iClass) {
				_itemResult.put(key, _contents.get(key));
			}
		}
		return _itemResult;
	}

	public void Remove(GenericItem item) {
		if (Contains(item)) {
			_contents.put(item, _contents.get(item) - 1);
			if (_contents.get(item) <= -1) {
				_contents.remove(item);
			}
		}
	}

	public boolean Contains(GenericItem item) {
		return _contents.containsKey(item);
	}

	public int GetItemCount(GenericItem item) {
		if (_contents.keySet().contains(item)) {
			return _contents.get(item);
		}
		return 0;
	}

	public void DropAll() {
		for (GenericItem item : _contents.keySet()) {
			while (_contents.get(item) > 0) {
				EntityManager.addObject(new GenericItem(item, _parent.GetLocation()));
				Remove(item);
			}
		}
	}

	float _nonZeroResult;

	public float NonZeroCount() {
		_nonZeroResult = 0;
		for (GenericItem item : _contents.keySet()) {
			if (_contents.get(item) > 0) {
				_nonZeroResult += _contents.get(item);
			}
		}
		return _nonZeroResult;
	}
}
